Enable probot autolabeler

alexvanboxel opened the pull request:
Waiting for INFRA enablement

R: iemejia

sunjincheng121 commented on .github/autolabeler.yml:
```diff
@@ -0,0 +1,26 @@
+python: ["sdks/python/*"]
+runner: ["runners/*"]
+examples: ["examples/*", "learning/*"]
+model: ["model/*"]
```
Add newline at the end of file.

alexvanboxel replied:
Added newline

iemejia commented on .github/autolabeler.yml:
```diff
@@ -0,0 +1,26 @@
+go: ["sdks/go/*"]
+java: ["sdks/java/*"]
+python: ["sdks/python/*"]
+runner: ["runners/*"]
```
Can we expand this also into the individual runners, e.g.
```
flink: ["runners/flink/*"]
spark: ["runners/spark/*"]
...
```

iemejia commented on .github/autolabeler.yml:
```diff
@@ -0,0 +1,26 @@
+java: ["sdks/java/*"]
+python: ["sdks/python/*"]
+runner: ["runners/*"]
+examples: ["examples/*", "learning/*"]
```
I would keep learning separate, these are different things and commonly mainted by different people who may be interesting in filtering one or the other.

iemejia commented:
I had almost the same suggestion than @sunjincheng121 however there is a small refinement to do over that one, do not put prefixes e.g. `extensions-java-sorter` use better `sorter` the other paths are matched by the other tag rules the specific rule for java and for extensions.

iemejia commented:
Forgot to mention, Can we sort the elements of the file so it can be easier to maintain too.

alexvanboxel commented:
I've added @iemejia and @sunjincheng121 suggestions. I did use a prefix though to make it usable in maintaining labels. I hope the auto-labeler can handle them. Prefix used are:

- extensions: ext/ (example: ext/jackson)
- io: io/ (example: io/mongodb)
- runners: run/ (example: run/flink)

I also sorted the labels. Squashed commit.

alexvanboxel commented:
Added the main labels already to the repo (not the sub labels)

iemejia commented:
Merged manually to do the extra adjustments. I did not want to bother anymore with those.
Since INFRA apparently enabled the action now is time to see those auto labels appearing :)
